partition_processor.go

```diff
@@ -350,7 +355,9 @@ func (pp *PartitionProcessor) run(ctx context.Context) (rerr error) {
 			// only fail processor if context not already Done
 			select {
 			case <-ctx.Done():
-				rerr = err
+				mutexErr.Lock()
+				rerr = newErrProcessing(pp.partition, fmt.Errorf("synchronous error in callback: %v", err))
```

Comment:
Nice catch. Technically it's quite likely that syncfailers will always be executed before, but it could happen that the asyncfailer fails first (e.g. if the consume-callback has a sleep?)
Anyway, it's fixed.